src/ol/source/imagestaticsource.js

```diff
@@ -26,10 +26,19 @@ ol.source.ImageStatic = function(options) {
 
   var imageExtent = options.imageExtent;
 
-  var resolution, resolutions;
+  var xResolution, yResolution, resolutions;
   if (options.imageSize !== undefined) {
-    resolution = ol.extent.getHeight(imageExtent) / options.imageSize[1];
-    resolutions = [resolution];
+    xResolution = ol.extent.getWidth(imageExtent) / options.imageSize[0];
+    yResolution = ol.extent.getHeight(imageExtent) / options.imageSize[1];
+    if (xResolution !== yResolution) {
+      if (xResolution > yResolution) {
+        resolutions = [xResolution, yResolution];
+      } else {
+        resolutions = [yResolution, xResolution];
```

Comment:
Yes, because this is a resolutions array, not the two resolutions that make up the aspect ratio. I think you can either use `Math.min` or `Math.max` here.